statsmodels/tsa/arima/specification.py

```diff
@@ -1047,6 +1047,161 @@ def construct_trend_names(self):
                 names.append('trend.%d' % i)
         return names
 
+    def validate_fixed_params(self, fixed_params):
+        """
+        Parameters
+        ----------
+        fixed_params : dict
+            Dictionary with names of fixed parameters as keys (e.g. 'ar.L1',
+            'ma.L2'), which correspond to SARIMAXSpecification.param_names.
+            Dictionary values are the values of the associated fixed
+            parameters.
+
+        Raises
+        ------
+        ValueError
+            If any parameters cannot be fixed, or
+
+        Notes
+        -----
+        Checks that 1) the parameter values are valid, and 2) the parameters
+        to be fixed are a subset of parameters that can be fixed
+        (i.e. param_names except "sigma2"). Please use `validate_params` for
+        validation on parameter values against contraints like
+        `enforce_stationarity=True`.
+
+        Examples
+        --------
+        >>> spec = SARIMAXSpecification(ar_order=1)
+        >>> spec.validate_params({"ar.L1": 1})  # valid
+        >>> spec.validate_params({"ar.L2": 1})
+        ValueError: Invalid fixed parameter(s): ['ar.L2'].
+        >>> spec.validate_params({"sigma2": 1})
+        ValueError: Invalid fixed parameter(s): ['sigma2'].
+        >>> spec.validate_params({"ar.L1": np.nan})
+        ValueError
+        """
+        assert isinstance(fixed_params, dict)
+
+        if len(fixed_params) == 0:
+            return
+
+        validate_basic(
+            list(fixed_params.values()), len(fixed_params),
+            allow_infnan=False, title='fixed parameters'
+        )
+
+        fixed_param_names = set(fixed_params.keys())
+        valid_param_names = set(self.param_names) - {"sigma2"}
```

Comment:
> We need to allow sigma2 to be fixed in the general `SARIMAXSpecification`, because this parameter can be fixed in the statespace and innovations MLE approaches (and also in principle in the other approaches, although I don't think we have any way to do that right now - it does not affect the coefficient estimates in most cases but it can affect the prediction intervals). So we will generally need to do this check in the specific estimation functions.

Ah, I see. What do you think about adding "allow_fixed_sigma2" argument here? Also, it's very likely I didn't understand this correctly, but I kind of thought `concentrate_scale` had the effect of fixing the `sigma2` (to 1). So on a related note, I also wonder how fixing `sigma2` should interact with `concentrate_scale`? As of now, it seems like we should only allow fixing `sigma2` when `concentrate_scale=False`. 

> (More generally, we have other checks on fixed parameters that will need to be done in specific estimation functions. For example, the default statespace case and innovations MLE case require the parameters to lead to a stationary process, and they impose this via a transformation function that transforms all parameters. As a consequence, in those cases we cannot, for example, fix the value of `ar.L2` while leaving the value of `ar.L1` free.)

What do you think about directly borrowing (or even migrating) from `SARIMAX._validate_can_fix_params`, which would remove the need for "estimator"-specific checks for most (possibly all) use cases?

Reply:
> Ah, I see. What do you think about adding "allow_fixed_sigma2" argument here?

I agree that this seems like a good approach, and implementing the check here should prevent boilerplate in the individual methods.

> I also wonder how fixing sigma2 should interact with concentrate_scale? As of now, it seems like we should only allow fixing sigma2 when concentrate_scale=False.

Yes I agree.  Currently in `SARIMAX`, if `concentrate_scale=True`, then `'sigma2'` is removed from the list of parameters, and so if you try to do e.g. `mod.fix_params({'sigma2': 4.3})`, you get: `ValueError: Invalid parameter name passed: "sigma2".`, so this is one option here, or you could be more explicit too.

> I kind of thought concentrate_scale had the effect of fixing the sigma2 (to 1)

You're right that this is practically what happens when the Kalman filter is being run, but at the same time it also computes an estimate of sigma2, and then that estimate is plugged back in after the fact. So the end result is a model that does not generally have sigma2 = 1.